pkg/httputil/config.go

```diff
@@ -0,0 +1,23 @@
+package httputil
+
+import (
+	"net/http"
+)
+
+var (
+	httpTransport *http.Transport
+)
+
+func AddTransport(transport *http.Transport) {
+	httpTransport = transport
+}
+
+func GetHttpClient() *http.Client {
+
+	if httpTransport != nil {
+		client := &http.Client{Transport: httpTransport}
```

Comment:
This will leak TCP sockets.  Should reuse the same client, similar to `http.DefaultClient`